Arduino/MiP_ProMini_Pack_Library/src/mip.cpp

```diff
@@ -1937,6 +1951,210 @@ int8_t MiP::rawGetUserData(uint8_t address, uint8_t& userData)
+void MiP::disableMiPDetectionMode()
+{
+    m_irId = MIP_IR_DETECTION_MODE_DISABLE;
+
+    // According to WowWee documentation, TX power must be between 1 and 120 even when disabling.
+    rawSetMiPDetectionMode(MIP_IR_DETECTION_MODE_DISABLE, 0x01);
+}
+
+bool MiP::isMiPDetectionModeEnabled()
+{
+    return m_irId > MIP_IR_DETECTION_MODE_DISABLE ? true : false;
+}
+
+uint8_t MiP::readDetectedMiP()
+{
+    // Fetch bytes from the Serial receive buffer and process any event data found within.
+    processAllResponseData();
+
+    uint8_t detectedMiPEvent = 0;
+    if(!m_detectedMiPEvents.pop(detectedMiPEvent))
+    {
+        m_lastError = MIP_ERROR_NO_EVENT;
+        return detectedMiPEvent;
+    }
+    m_lastError = MIP_ERROR_NONE;
+    return detectedMiPEvent;
+}
+
+uint8_t MiP::availableDetectedMiPEvents()
+{
+    // Fetch bytes from the Serial receive buffer and process any event data found within.
+    processAllResponseData();
+    m_lastError = MIP_ERROR_NONE;
+    return m_detectedMiPEvents.available();
+}
+
+// This internal protected method sends the set detection mode command with minimal error
+// handling. The error recovery happens at a higher level of the driver.
+void MiP::rawSetMiPDetectionMode(uint8_t id, uint8_t txPower)
+{
+    uint8_t command[1+2];
+
+    MIP_ASSERT( 0x01 <= txPower && txPower <= 0x78 );
+
+    command[0] = MIP_CMD_SET_DETECTION_MODE;
+    command[1] = id;
+    command[2] = txPower;
+
+    rawSend(command, sizeof(command));
+}
+
+void MiP::enableIRRemoteControl()
+{
+    verifiedIRRemoteControl(MIP_IR_REMOTE_CONTROL_ENABLE);
+}
+
+void MiP::disableIRRemoteControl()
+{
+    verifiedIRRemoteControl(MIP_IR_REMOTE_CONTROL_DISABLE);
+}
+
+bool MiP::isIRRemoteControlEnabled()
+{
+    const uint8_t remoteControlEnabled[1] = { MIP_CMD_GET_IR_REMOTE_CONTROL };
+    uint8_t       response[1+1];
+    size_t        responseLength;
+    int8_t        result;
+
+    result = rawReceive(remoteControlEnabled, sizeof(remoteControlEnabled), response, sizeof(response), responseLength);
+
+    if (result)
+    {
+        return result;
+    }
+    if (responseLength != sizeof(response) ||
+        response[0] != MIP_CMD_GET_IR_REMOTE_CONTROL)
+    {
+        return MIP_ERROR_BAD_RESPONSE;
+    }
+
+    return response[1] == MIP_IR_REMOTE_CONTROL_ENABLE ? true : false;
+}
+
+void MiP::sendIRDongleCode(uint8_t sendCode[], uint8_t transmitPower)
+{
+    uint8_t command[1+6];
+    command[0] = MIP_CMD_SEND_IR_DONGLE_CODE;
+    command[1] = 0x00;
+    command[2] = 0x00;
+    command[3] = sendCode[0];
+    command[4] = sendCode[1];
+    command[5] = 0x10;
+    command[6] = transmitPower;
+
+    // Send this command blindly with no error checking since there is no robust way to determine if it has failed.
+    rawSend(command, sizeof(command));
+}
+
+uint32_t MiP::readIRDongleCode()
+{
+    // Fetch bytes from the Serial receive buffer and process any event data found within.
+    processAllResponseData();
+
+    uint32_t irCodeEvent;
```

Comment:
Maybe initialize it to 0xFFFFFFFF here, incase the queue is empty.

Reply:
Pending amended commit.